apps/blockchain/lib/eth_common_test/helpers.ex

```diff
@@ -109,4 +111,100 @@ defmodule EthCommonTest.Helpers do
   def ethereum_common_tests_path do
     Path.join(System.cwd(), "/../../ethereum_common_tests")
   end
+
+  @spec known_failures_path :: String.t()
+  def known_failures_path do
+    Path.join(System.cwd(), "/test/support/known_failures.txt")
+  end
+
+  @spec run_common_tests(String.t(), (String.t(), %{} -> no_return())) :: no_return()
+  def run_common_tests(test_set_name, runner) do
+    # `common_tests` are simply a list of paths to tests for test_set_name
+    # these are top-level folder names in `ethereum_common_tests`
+    common_tests = EthCommonTest.Helpers.test_files(test_set_name)
+
+    known_failures =
+      case File.read(known_failures_path()) do
```

Comment:
maybe we should use `File.read!`. Because current pattern matching can potentially skip files if a wrong path is provided

Reply:
The idea was that we always look for "known_failures.txt" and if it's not there, then we expect all tests to pass. In other words, the default is to run all tests, but you can include `known_failures.txt` if some common tests are not yet passing.

Reply:
(Part of the reason for this is that one of the tests was ignoring a common test and it wasn't obvious until we dug into the file that it was skipping a core test. This is meant to hang a lampshade on the issues).